src/quo2/components/community/token_gating.cljs

```diff
@@ -0,0 +1,168 @@
+             :header-info-button {:margin-left 8}
+             :token-requirement-list-spacing {:padding-vertical 8}
+             :token-list {:align-items :center}
+             :token-row {:flex-direction :row}
+             :token-tag-spacing {:margin-left 7}
+             :divider {:height 1
+                       :margin-horizontal -20
+                       :margin-top 12
+                       :margin-bottom 8}
+             :enter-button {:margin-top 16}
+             :info-text {:margin-top 12
+                         :text-align :center
+                         :padding-horizontal 20}})
+
+(defn is-token-requirement-met? [token-requirement-list]
+  (every?
+   (fn [token]
+     (get token :is-sufficient?))
+   token-requirement-list))
+
+(defn token-requirement-list [props]
+  (let [{:keys [gate]} props
+        [gate-type token-requirement-list] gate
+        is-sufficient? (is-token-requirement-met? token-requirement-list)
+        not-text (if-not is-sufficient? " not" "")
+        message (if
+                 (= gate-type :join)
+                  (i18n/label
+                   :t/community-join-requirements {:not not-text})
+                  (if (= gate-type :read)
+                    (i18n/label
+                     :t/community-channel-read-requirements {:not not-text})
+                    (i18n/label
+                     :t/community-channel-write-requirements {:not not-text})))]
+    [rn/view
+     [rn/view {:style (get styles :token-requirement-list-spacing)}
+      [text/text {:weight :medium
+                  :number-of-lines 1} message]
+      [text/text {:size :paragraph-2
+                  :number-of-lines 1} "You must hodl:"]]
+
+     [rn/view {:style (merge (get styles :token-row) (merge (get styles :token-requirement-list-spacing)) (get styles :token-list))}
+      (map-indexed (fn [token-index token]
+                     ^{:key token}
+                     [:<>
+                      [rn/view {:style (when-not (= token-index 0) (get styles :token-tag-spacing))}
+                       [token-tag/token-tag {:token (:token token)
+                                             :value (:amount token)
+                                             :size :big
+                                             :is-required (:is-sufficient? token)
+                                             :is-purchasable (:is-purchasable token)
+                                             :token-img-src
+                                             (js/require
+                                              (str
+                                               "../resources/images/tokens/mainnet/"
+                                               (:token token)
+                                               ".png"))}]]])
+                   token-requirement-list)]]))
+
+(defn is-community-locked? [community]
+  (let [{gates :gates} community]
+    (when (some? gates)
+      (when (vector? (:join gates))
```

Comment:
you can just use `(when (vector? (:join gates))` only, `(when (some? gates)` is redundant

Reply:
thanks!